ui/src/app/views/homepage/component.spec.ts

```diff
@@ -38,9 +38,13 @@ import {RightScrollLightComponent} from 'app/icons/right-scroll-light/component'
 import {RightScrollComponent} from 'app/icons/right-scroll/component';
 import {ShrinkComponent} from 'app/icons/shrink/component';
 
+import * as React from 'react';
+import * as ReactTestUtils from 'react-dom/test-utils'
```

Comment:
We generally avoid * imports in java - I would think we should keep that tradition here for the same reason.

Reply:
so we found that this was the only way to get react to properly import for tests to be able to run.  when we import just as "import React from 'react'" we get this error:
```
ERROR in src/app/views/quick-tour-modal/component.tsx(2,8): error TS1192: Module '"/w/ui/node_modules/@types/react/index"' has no default export.
src/app/views/quick-tour-modal/component.tsx(3,8): error TS1192: Module '"/w/ui/node_modules/@types/react-dom/index"' has no default export.
```

and in the test suite:
```
Uncaught TypeError: Cannot read property 'createElement' of undefined
```

Reply:
FYI: on that particular error message, `import {React} from 'react'` *might* get you further (no promises).